### PR TITLE
CLI: Add option "cardano-wallet transaction create --metadata=JSON"

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-307 / #2076

### Overview

- Adds `--metadata=JSON` option to transaction create and fee estimate commands.

### Comments

Will update the CLI wiki page after merging.

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/CLI/Shelley/Transactions.hs [hunk 1]
```diff
@@ -294,6 +300,36 @@ spec = do
         out `shouldBe` ""
         c `shouldBe` ExitFailure 1
 
+    it "TRANSMETA_CREATE_01 - Transaction with metadata via CLI" $ \ctx -> do
+        (wSrc, wDest) <- (,) <$> fixtureWallet ctx <*> emptyWallet ctx
+        let amt = 100000
+        let md = Just "{ \"1\": \"hello\" }"
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Scenario/CLI/Shelley/Transactions.hs]: :wave:

**Review by KtorZ — APPROVED** [review]
:ok_hand:

**Comment by rvl** [comment]
Oops - the integration test transaction did not exceed minimum UTxO.

bors r+

**Comment by rvl** [comment]
Should be updated now.

bors r+

**Comment by KtorZ** [comment]
bors merge

**Comment by rvl** [comment]
@KtorZ You force pushed over my other build fixes. :slightly_smiling_face: I have pushed them back, rebased, and updated for #2147.

bors r+

**Pull request merged**